js/packages/wallet-adapter-mobile/src/adapter.ts

```diff
@@ -233,6 +221,33 @@ export class SolanaMobileWalletAdapter extends BaseMessageSignerWalletAdapter {
         });
     }
 
+    async performAuthorization(signInPayload?: SignInPayload): Promise<AuthorizationResult> {
+        try {
+            const cachedAuthorizationResult = await this._authorizationResultCache.get();
+            if (cachedAuthorizationResult) {
+                // TODO: Evaluate whether there's any threat to not `awaiting` this expression
+                this.handleAuthorizationResult(cachedAuthorizationResult);
+                return cachedAuthorizationResult;
+            }
+            return await this.transact(async (wallet) => {
+                const authorizationResult = await wallet.authorize({
```

Comment:
What happens if a SIWS request occurs but the mobile wallet doesn't support SIWS? 

Would the `wallet.authorize` call throw an error (good) or simply ignore the signInPayload and use normal `authorize`?